app/api/auth.py

```diff
@@ -58,6 +58,10 @@ def authenticate(allow_refresh_token=False, existing_identity=None):
     password = data.get('password')
     criterion = [username, password]
 
+    user = User.query.filter_by(_email=username).first()
+    if user.is_blocked:
+        return jsonify(error='Admin has marked this account as Spam'), 400
```

Comment:
Wrong, this is leaking info about user being banned. I can now know if a user is banned or not by just knowing their email. Think more

Reply:
```
 user = User.query.filter_by(_email=username, password=password).first()
    if user and user.is_blocked:
        return jsonify(error='Admin has marked this account as Spam'), 400
```

is this ok? it will require username and password both.

Reply:
See the entire function, what you've written above is already being done somewhee